nix/haskell.nix

```diff
@@ -51,56 +51,39 @@ let
       modules =
-        let
-          inherit (config) src;
-          # deduce package names and exes from the cabal project to avoid hard-coding them:
-          projectPackagesExes =
-            let
-              project = haskell-nix.cabalProject' (builtins.removeAttrs config [ "modules" ]);
-              packages = haskellLib.selectProjectPackages project.hsPkgs;
-            in
-            lib.genAttrs
-              (lib.attrNames packages)
-              (name: lib.attrNames packages.${name}.components.exes);
-          projectPackageNames = builtins.attrNames projectPackagesExes;
-        in
         [
-          ({ pkgs, ... }: {
+          ({ lib, pkgs, ... }: {
             packages.cardano-tracer.package.buildable = with pkgs.stdenv.hostPlatform; lib.mkForce (!isMusl);
             packages.cardano-node-chairman.components.tests.chairman-tests.buildable = lib.mkForce pkgs.stdenv.hostPlatform.isUnix;
             packages.plutus-tx-plugin.components.library.platforms = with lib.platforms; [ linux darwin ];
             packages.tx-generator.package.buildable = with pkgs.stdenv.hostPlatform; !isMusl;
           })
-          ({ pkgs, ... }: {
+          ({ lib, pkgs, ... }: {
             # Needed for the CLI tests.
             # Coreutils because we need 'paste'.
             packages.cardano-testnet.components.tests.cardano-testnet-tests.build-tools =
               lib.mkForce (with pkgs.buildPackages; [ jq coreutils shellcheck lsof ]);
           })
-          ({ pkgs, ... }: {
+          ({ lib, pkgs, ... }: {
             # Use the VRF fork of libsodium
             packages.cardano-crypto-praos.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf ] ];
             packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 ] ];
           })
-          ({ pkgs, options, ... }: {
+          ({ lib, pkgs, ... }: lib.mkIf (!pkgs.stdenv.hostPlatform.isWindows) {
             # add shell completion:
-            packages = lib.mapAttrs
-              (name: exes: {
-                components.exes = lib.genAttrs exes (exe: {
-                  postInstall = lib.optionalString
-                    (!pkgs.stdenv.hostPlatform.isWindows
-                      && lib.elem exe [ "cardano-node" "cardano-cli" "cardano-topology" "locli" ]) ''
-                    BASH_COMPLETIONS=$out/share/bash-completion/completions
-                    ZSH_COMPLETIONS=$out/share/zsh/site-functions
-                    mkdir -p $BASH_COMPLETIONS $ZSH_COMPLETIONS
-                    $out/bin/${exe} --bash-completion-script ${exe} > $BASH_COMPLETIONS/${exe}
-                    $out/bin/${exe} --zsh-completion-script ${exe} > $ZSH_COMPLETIONS/_${exe}
-                  '';
-                });
-              })
-              projectPackagesExes;
+            packages.cardano-node.components.exes = 
+              lib.genAttrs [ "cardano-node" "cardano-cli" "cardano-topology" "locli" ]
```

Comment:
I don't think those exes are all in the `Cardano-node` package.

Reply:
That's interesting because I checked that this PR would not change the final derivation. Try yourself

```
nix-diff \
    $(nix eval --raw github:input-output-hk/cardano-node/7bf09ba8038bda4b3bcba2d2344afc0346ea7a46 --apply 'x: x.drvPath') \
    $(nix eval --raw github:input-output-hk/cardano-node/8c1b6ce28df3a2bd33946c5912df6e3d210bd856 --apply 'x: x.drvPath')
```

I might need to investigate

Reply:
I think that only compares the default derivation package of the flake (which is the `cardano-node` exe derivation).  The others are probably used in tests and shells.  Comparing `.#hydraJobs.required` might show differences.

Reply:
I think this was already broken when we switched to using Cardano-cli from CHaP.  That probably required something like:

```diff
diff --git a/nix/haskell.nix b/nix/haskell.nix
index 4df3c76fe..831094dd4 100644
--- a/nix/haskell.nix
+++ b/nix/haskell.nix
@@ -98,7 +98,7 @@ let
                   '';
                 });
               })
-              projectPackagesExes;
+              (projectPackagesExes // { cardano-cli = ["cardano-cli"]; });
           })
           ({ pkgs, config, ... }:
             let
```

Reply:
This PR breaks `cardano-toplology`:

```
$ nix-diff $(nix path-info --derivation github:input-output-hk/cardano-node/7bf09ba8038bda4b3bcba2d2344afc0346ea7a46#cardano-topology) $(nix path-info --derivation github:input-output-hk/cardano-node/8c1b6ce28df3a2bd33946c5912df6e3d210bd856#cardano-topology)
- /nix/store/wpk68gl3vn0ha6vy4qk8zp4vllwm3d2n-cardano-topology-exe-cardano-topology-8.1.0.drv:{out}
+ /nix/store/44n7zcxg5l9bqz2bpycrsrn3mxa5y46d-cardano-topology-exe-cardano-topology-8.1.0.drv:{out}
• The environments do not match:
    - postInstall=BASH_COMPLETIONS=$out/share/bash-completion/completions
ZSH_COMPLETIONS=$out/share/zsh/site-functions
mkdir -p $BASH_COMPLETIONS $ZSH_COMPLETIONS
$out/bin/cardano-topology --bash-completion-script cardano-topology > $BASH_COMPLETIONS/cardano-topology
$out/bin/cardano-topology --zsh-completion-script cardano-topology > $ZSH_COMPLETIONS/_cardano-topology
```

Reply:
:+1: fixed now. Also I have added the postInstall to cardano-cli, which has you noticed was gated behind `isLocalPackage` while it's not a local package anymore.